src/components/panel/panel.js

```diff
@@ -784,22 +897,71 @@ angular
 
 
 // Default z-index for the panel.
-var defaultZIndex = 80;
+var MD_PANEL_Z_INDEX = 80;
 var MD_PANEL_HIDDEN = '_md-panel-hidden';
-
 var FOCUS_TRAP_TEMPLATE = angular.element(
     '<div class="_md-panel-focus-trap" tabindex="0"></div>');
 
 
 /**
+ * A provider that is used for building configurations for the panel.
+ * @final @constructor @ngInject
+ */
+function MdPanelConfigProvider() {
+  /** @private {!Object} */
+  this._configs = {};
+}
+
+
+/**
+ * Takes the passed in panel configuration object and adds it to the `_configs`
+ * object of the provider.
+ * @param {!Object<string, !Object>} config Specific configuration object that
+ *     is required to contain a name string and an options object. The options
+ *     object can contain any and all of the parameters available within the
+ *     `$mdPanel.create` configuration. However, it does not make sense to
+ *     include parameters that depend on certain interactions of the user,
+ *     positioning, or animation.
+ */
+MdPanelConfigProvider.prototype.createConfig = function(config) {
```

Comment:
I think `createConfig` should take two params to help avoid the confusion with fields should be in the config. You're creating one at a time so it really shouldn't matter.

Also, consider a builder if you want to chain the creates.

```createConfig(name, config)```